monitoring/inotify_linux_test.go

```diff
@@ -0,0 +1,37 @@
+/*
+Copyright 2018 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package monitoring
+
+import (
+	"context"
+
+	"github.com/gravitational/satellite/agent/health"
+	"gopkg.in/check.v1"
+)
+
+type InotifySuite struct{}
+
+func (_ *InotifySuite) TestCreateInotify(c *check.C) {
+	// Note, it's difficult to simulate on linux
```

Comment:
Not if you abstract `InotifyInit` and `InotifyAddWatch` into `Init` and `AddWatch` and simply simulate whatever conditions you're interested in.

Reply:
Not sure what you mean by 18 line function becoming completely unreadable with the use of abstractions instead of real syscalls. If you had it, you could mock whatever results you needed - you're not using that much of the functionality to not be able to do that.